core/src/com/marama/view/entities/EntityManagerSingleton.java

```diff
@@ -0,0 +1,82 @@
+package com.marama.view.entities;
+
+import com.badlogic.gdx.assets.AssetManager;
+import com.badlogic.gdx.graphics.g3d.loader.ObjLoader;
+import com.badlogic.gdx.utils.ObjectMap;
+import com.marama.view.entities.exceptions.ModelNotFoundException;
+import com.marama.view.entities.instances.SelectableInstance;
+import com.marama.view.util.MaramaficationLoader;
+import com.sun.media.jfxmedia.logging.Logger;
+
+/**
+ * The EntityManager singleton class, extending from {@link AssetManager}.
+ */
+public class EntityManagerSingleton extends AssetManager {
+    private static EntityManagerSingleton instance = new EntityManagerSingleton();
+
+    // Since in AssetManager there is no good way to get all assets of a certain type, we have to remember our maramafications here.
+    final ObjectMap<String, Maramafication> maramafications = new ObjectMap();
+
+    public ObjLoader objLoader = new ObjLoader();
+
+    /**
+     * An instance of {@link AssetManager}, but it's also able to load maramafications.
```

Comment:
An instance of Assetmanager that is also able to load maramafications.

Might be better?

Reply:
Sounds better, on top of this, I don't think you should put the name of the design pattern in the class name. I would just call it EntityManager. The description/javaDoc says enough about what this class is and does.